preprocess.py

```diff
@@ -0,0 +1,205 @@
+import datetime
+import functools
+import json
+import logging
+import os
+from collections import defaultdict
+
+import coiled
+import deltalake
+import fsspec
+import pandas as pd
+import toolz
+from dask.distributed import LocalCluster, print, wait
+from upath import UPath as Path
+
+logger = logging.getLogger(__name__)
+
+LOCAL = False
+if LOCAL:
+    OUTDIR = Path(f"./data")
+    STORAGE_OPTIONS = {}
+    Cluster = LocalCluster
+else:
+    OUTDIR = Path(f"s3://openscapes-scratch/etl-github")
+    STORAGE_OPTIONS = {"AWS_REGION": "us-west-2", "AWS_S3_ALLOW_UNSAFE_RENAME": "true"}
+    Cluster = functools.partial(
+        coiled.Cluster,
+        name="etl-github",
+        n_workers=50,
+        region="us-west-2",
+        shutdown_on_close=False,
```

Comment:
Note I'm keeping the cluster up for rapid iteration on the script. Not a long term thing.